### ISSUE TITLE
Offering to maintain nps

### ISSUE THREAD

**Issue by mf-bruce** [sezna/nps#234]
First off, thanks @sezna for building nps in the first place! We have been using it at Markforged for some time now, and have found it to be elegant in its simplicity, whilst providing just enough functionality to be incredibly powerful!

It seems that nps has lacked maintenance for some years now, there are outstanding [security issues](https://github.com/sezna/nps/issues/232), and feature requests such as [positional args](https://github.com/sezna/nps/issues/177) (something we have [implemented in our fork](https://github.com/MarkForged/nps/commit/e99f1a9496019f0064bf5106056c46f40294d1e6)).

We would like to offer to maintain the `nps` package, so opening this issue to start that conversation!

**Comment by sezna** [comment]
Yeah, I took over because my previous job was using it and it wasn't being maintained. Now, I'm no longer in that role and I haven't had the time to give it any love. Happy to find somebody who is willing to give it the love it needs!

**Comment by mf-bruce** [comment]
Awesome! How do we go about transferring this? Would it make sense to transfer the npm package ownership so we can publish from our repo moving forward?

**Comment by Melmoth-the-Wanderer** [comment]
@mf-bruce have you taken over the project?

**Comment by mf-bruce** [comment]
Yep, you can find our fork
* on [Github](https://github.com/MarkForged/nps)
* on [npm](https://www.npmjs.com/package/@markforged/nps) as `@markforged/nps`

Would be nice to transfer the original package name on npm, or at least deprecate it. FYI it's possible the npm deploy key for the original package was leaked via travis:( https://arstechnica.com/information-technology/2021/09/travis-ci-flaw-exposed-secrets-for-thousands-of-open-source-projects/

**Comment by mf-bruce** [comment]
@sezna the current status quo will work fine for folks who proactively want to use a newer version of `nps` but there are some things we can do to smooth the transition for users who just have "latest" specified in package.json, or new folks discovering this package. Raising these here as I will need some assistance from yourself if we are going to execute on them.

* Transfer issues to our fork
** I would need write access to this repo to transfer issues)
* Set our fork as the root repo for other forks
** I asked github support about this and they need you to additionally submit a ticket which references ours
* Archive this fork
* Transfer ownership of the [nps npm package](https://www.npmjs.com/package/nps) to us

**Comment by nareshbhatia** [comment]
I just stumbled upon this package today and found it very interesting. But looking at the last commit almost 2 years ago, was hesitant to pick it up. Until I saw this issue. Great job @mf-bruce to keep it going.

I would personally like to see mf-bruces fork merged over here so that star ratings can be maintained. There is lot of equity build into the stars and the downloads - would be sad to loose it all. And also the complication of moving the issues and all. @senza, in the best interest of the community it may be best to merge mf-bruce's fork here and give him permissions to publish on npm. You both can still co-own the package.

P.S. I also see @kentcdodds on the contributor's list, so this has to be good :-)

**Comment by anodynos** [comment]
I would love to use nps as well, especially since [JSON5 is out of the question for package.json](https://github.com/npm/feedback/discussions/56), but I'm hesitant after seeing trivialities [like these](https://github.com/MarkForged/nps/issues/12) hanging for years ;-(

**Comment by sezna** [comment]
Hey Naresh, yeah, I thought I had time for this many years ago but I haven't. We used it at my last company, and it was going unmaintained, so I took over. But, here I am, in the same situation. 

It looks like Bruce has left Markforged, though, so I'm not sure who a good contact would be. And it looks like their fork isn't a "fork" in that I can't merge from it. It also hasn't seen much activity. 

If anybody would be interested in maintaining this package, I'd be happy to transfer it.

**Comment by anodynos** [comment]
I think I might be interested, but I'm engaged with other projects for the next few months... I will let you know...

**Comment by JounQin** [comment]
@sezna I'm interested to take over like the one I did at https://github.com/prettier/pretty-quick/issues/164#issuecomment-1895400862

We're using `nps` at https://github.com/prettier/prettier-eslint, so I want it be maintained actively.

**Comment by sezna** [comment]
Hey @JounQin I'd appreciate that. You seem to be pretty involved in the JS community so no issues on my part. I've sent an email to the address listed on your profile.

**Comment by JounQin** [comment]
@sezna

Hi, I reponded with your email yesterday but I don't know if you received in case that's a private email domain, so that could be listed in your spam folder, so I'm trying to reply with GitHub comment here.

Quote my original response:

---

Thanks for the agreement!
 
My npm username is [jounqin](https://www.npmjs.com/~jounqin), for the repository I don't know if it can be transformed from you to me directly, or maybe a 3rd party organization is needed.
So you can give it a try, if it doesn't work for transform directly, I can invite you into @un-es for example, and then you should be able to transform that reo into the org, then all done.
 
Finally, if you got a social account which posts `nps` previously, a tweat about the ownership transformation would be good for the users I think, but of course it's not required or if you don't like it.
 
Anyway, thanks so much for this!

Best Wishes -- JounQin

**Comment by JounQin** [comment]
Friendly ping @sezna .

**Comment by JounQin** [comment]
Friendly ping @sezna.

**Comment by JounQin** [comment]
Friendly ping @sezna.

**Comment by JounQin** [comment]
Friendly ping @sezna.
